### PR TITLE
feat(typeahead): add support for the 'container' input

### LINKED ISSUE #1105: Typeahead: support like 'append-to-body'
### Bug description: Feature Request

I hope to support like `typeahead-append-to-body` of ui-bootstrap(for Angular1)
because typeahead window in modal window is clipped.

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 2.2.4

ng-bootstrap: 1.0.0.beta-14

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Closes #1105

### CODE REVIEW

File: src/typeahead/typeahead.ts [hunk 1]
```diff
@@ -91,6 +91,12 @@ export class NgbTypeahead implements ControlValueAccessor,
 
 
   /**
+   * A selector specifying the element the tooltip should be appended to.
+   * Currently only supports "body".
+   */
+  @Input() container: string;
```
Comment (maxokorokov) [inline src/typeahead/typeahead.ts]: 1. Can't we limit it to `@Input() container: 'body';` in this case?
2. You should probably add it to the `NgbTypeaheadConfig`

File: src/typeahead/typeahead.ts [hunk 2]
```diff
@@ -241,6 +250,10 @@ export class NgbTypeahead implements ControlValueAccessor,
       this._windowRef.instance.id = this.popupId;
       this._windowRef.instance.selectEvent.subscribe((result: any) => this._selectResultClosePopup(result));
       this._windowRef.instance.activeChangeEvent.subscribe((activeId: string) => this.activeDescendant = activeId);
+
+      if (this.container === 'body') {
+        window.document.querySelector(this.container).appendChild(this._windowRef.location.nativeElement);
```
Comment (wesleycho) [inline src/typeahead/typeahead.ts]: We probably want to inject DOCUMENT and Renderer2 and make use of those

Comment (pkozlowski-opensource) [inline src/typeahead/typeahead.ts]: Current `Renderer2` interface doesn't have any method(s) to query nodes and / or call an arbitrary method on a DOM node: https://github.com/angular/angular/blob/10897d64738484036443f6614ad3cbef095da0d5/packages/core/src/render/api.ts#L147-L191

File: src/typeahead/typeahead.ts (hunk 1 above)
Comment (pkozlowski-opensource) [inline src/typeahead/typeahead.ts]: > Can't we limit it to @Input() container: 'body'; in this case?

I've modeled the proposed API after all the other components that have the `container` option (tooltip, popover etc.). We can change it across the board in a separate PR.

> You should probably add it to the NgbTypeaheadConfig

Yup, good idea 👍

**Pull request closed without merging**